### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by Ibtisam34**
- [ ]  In this step, I configure my Redux Store and write actions and reducer for adding and removing books.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,9 +9,13 @@
...
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-redux": "^8.0.5",
     "react-router-dom": "^6.5.0",
     "react-scripts": "5.0.1",
     "react-svg": "^15.1.16",
+    "redux": "^4.2.0",
+    "redux-logger": "^3.0.6",
+    "uuid": "^9.0.0",
     "web-vitals": "^2.1.4"
```
Comment (sja-thedude) [inline package.json:20]: - [ ] As per the project requirements kindly make sure to Add Redux Toolkit(npm install react-redux @reduxjs/toolkit).

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import booksReducer from './books/books';
+
+const reducer = combineReducers({
+  booksReducer,
+});
+
+const store = createStore(
+  reducer,
+  applyMiddleware(logger),
+);
```
Comment (sja-thedude) [inline src/redux/configureStore.js]: - [ ] Kindly add categories in configureStore.js, the way you have added books with combining reducers.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Ibtisam34, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters passing
✔️ Professional Readme
✔️ Correct Gitflow

## Required 
- [ ] Kindly make sure to follow the duck's pattern and add a categories folder in a redux folder as shown. Currently, you are missing it.
Screenshot 2023-01-02 at 12 35 16 PM

- [ ] Kindly Write your categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux) as given in the project requirements.
Screenshot 2023-01-02 at 12 35 45 PM

- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export const checkStatus = (category) => ({
+  type: 'check Status',
+  payload: category,
+});
+
+const categoryReducer = (state = [], action) => {
+  switch (action.type) {
+    case 'check status':
+      return 'Under construction';
```
Comment (TedLivist) [inline src/redux/categories/categories.js]: - [ ] Kindly add the action type and refactor your action creator in keeping with the standard. See example:

  ```
  const CHECK_STATUS = 'bookstore/categories/CHECK_STATUS';
  const initialState = [];
  
  export const checkStatus = () => ({
    type: CHECK_STATUS,
  });
  
  const categoryReducer = (state = initialState, action) => {
    switch (action.type) {
      case CHECK_STATUS:
        return 'Under Construction';
  ...
  ```

File: src/redux/configureStore.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  booksReducer,
+  category: categoryReducer,
+});
+
+const store = createStore(
+  reducer,
+  applyMiddleware(logger),
+);
+
+export default store;
```
Comment (TedLivist) [inline src/redux/configureStore.js]: - [ ] Good job installing the `@reduxjs/toolkit` package in the `package.json` file. However, you need to import `configureStore` from it to configure your store. Kindly refactor accordingly. See example:

  ```
  import { combineReducers } from 'redux';
  import { configureStore } from '@reduxjs/toolkit';
  import booksReducer from './books/books';
  import categoryReducer from './categories/categories';
  
  const reducer = combineReducers({
    books: booksReducer,
    categories: categoryReducer,
  });
  
  const store = configureStore({
    reducer,
  });
  
  export default store;
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Ibtisam34,

✔️ Implementation of ducks pattern

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categories.js (hunk 3 above)
Comment (Ibtisam34) [reply src/redux/categories/categories.js]: kindly plz Iam not understanding this section what exactly you want to see https://github.com/TedLivist

Comment (TedLivist) [reply src/redux/categories/categories.js]: Hi @Ibtisam34 I mean to be consistent
  - You should have an action type:
    ```
    const CHECK_STATUS = 'bookstore/categories/CHECK_STATUS'
    ```
  - And refactor your action creator (checkStatus) to use the action type you defined:
    ```
    export const checkStatus = () => ({
      type: CHECK_STATUS,
    });
    ```
  - Then refactor the reducer accordingly.

**Review by OLIPLICHE — APPROVED** [review]
Hi @Ibtisam34 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Implemented the changes requested from the previous review👏 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**